dist/screeps.d.ts

```diff
@@ -1017,8 +1017,7 @@ interface SignDefinition {
 }
 interface StoreDefinition {
     [resource: string]: number;
-    energy?: number;
-    power?: number;
+    energy: number;
```

Comment:
>Is this actually the case, is energy defined as true all the time?

Yes.

The manual documents it separately for each of the different things which use a StoreDefinition, eg:
http://support.screeps.com/hc/en-us/articles/208435885-StructureContainer#store

>An object with the structure contents. Each object key is one of the RESOURCE_* constants, values are resources amounts. RESOURCE_ENERGY is always defined and equals to 0 when empty, other resources are undefined when empty.


As for "other resources are undefined when empty." - this is ambiguously worded. It could be taken to mean that other resources are absent from the object when undefined, or that they may be present, but set to `undefined`. Empirical testing suggests it's always the former, meaning we can avoid having `| undefined` in our typing and slightly simplify everyone's code.